library/Process-linux.cpp

```diff
@@ -120,13 +121,18 @@ Process::~Process()
 
 string Process::doReadClassName (void * vptr)
 {
-    //FIXME: BAD!!!!!
-    char * typeinfo = Process::readPtr(((char *)vptr - sizeof(void*)));
-    char * typestring = Process::readPtr(typeinfo + sizeof(void*));
+    char* typeinfo = Process::readPtr(((char *)vptr - sizeof(void*)));
+    char* typestring = Process::readPtr(typeinfo + sizeof(void*));
     string raw = readCString(typestring);
-    size_t start = raw.find_first_of("abcdefghijklmnopqrstuvwxyz");// trim numbers
-    size_t end = raw.length();
-    return raw.substr(start,end-start);
+
+    string status;
+    string demangled = cxx_demangle(raw, &status);
```

Comment:
does `devel/scan-vtables` produce the same output as before (on Linux) with this change?

Reply:
scan-vtables doesn't use doReadClassName, and instead uses cxx_demangle directly from lua. I haven't adjusted the linux implementation of that function.